R/copy-model-from.R

```diff
@@ -222,17 +222,16 @@ copy_control_stream <- function(.parent_model_path, .new_model_path, .overwrite,
   if (.update_model_file) {
 
     # read parent control stream
-    mod_str <- .parent_model_path %>% read_file()
+    mod_str <- nmrec::read_ctl(.parent_model_path)
 
     # replace the $PROBLEM line(s)
     new_mod_id <- get_model_id(.new_model_path)
-    mod_str <- str_replace(mod_str,
-                           "\\$PROB(.|\n)*?\\$",
-                           as.character(glue("$PROBLEM From bbr: see {new_mod_id}.yaml for details\n\n$")))
-    glue("")
+    prob_rec <- nmrec::select_records(mod_str, "prob")[[1]]
+    prob_str <- nmrec::get_record_option(prob_rec, "text")
```

Comment:
As we discussed a bit offline, there's a wrinkle here in that NONMEM will accept `$PROB{newline}` as a valid record. [61c85f7](https://github.com/metrumresearchgroup/nmrec/commit/61c85f7a0996df46b0240c59a59ec841d1b26864) (define parser for problem records, 2023-06-30) explains the situation a bit.

In this case, there won't be a `text` option. At the very least, this spot should detect a `NULL` return from `get_record_option`, stop trying to update it, and warn the user.

A more involved fix could be something like this (again, untested):

```diff
diff --git a/R/copy-model-from.R b/R/copy-model-from.R
index 584ca3b3..266fe5b9 100644
--- a/R/copy-model-from.R
+++ b/R/copy-model-from.R
@@ -227,8 +227,21 @@ copy_control_stream <- function(.parent_model_path, .new_model_path, .overwrite,
     # replace the $PROBLEM line(s)
     new_mod_id <- get_model_id(.new_model_path)
     prob_rec <- nmrec::select_records(mod_str, "prob")[[1]]
+    new_text <- as.character(glue("From bbr: see {new_mod_id}.yaml for details"))
+
     prob_str <- nmrec::get_record_option(prob_rec, "text")
-    prob_str$value <- as.character(glue("From bbr: see {new_mod_id}.yaml for details"))
+    if (is.null(prob_str)) {
+      # NONMEM permits a bare "$PROB{newline}", in which case there won't be a
+      # text option.
+      prob_rec$parse()
+      idx <- purrr::detect_index(
+        prob_rec$values,
+        function(x) inherits(x, "nmrec_option_record_name")
+      )
+      prob_rec$values <- append(prob_rec$values, paste0(" ", new_text), idx)
+    } else {
+      prob_str$value <- new_text
+    }

     # write parent control stream
     nmrec::write_ctl(mod_str, .new_model_path)
```

(That could be a bit more careful and signal a dev error if `idx == 0`.)